src/main/scala/it/pagopa/interop/catalogprocess/common/readmodel/ReadModelCatalogQueries.scala

```diff
@@ -193,8 +200,28 @@ object ReadModelCatalogQueries extends ReadModelQuery {
 
     val modeFilter = mode.map(_.toString).map(Filters.eq("data.mode", _))
 
+    /**
+      * Allow only the producer with right permissions to see eservices in draft, that is:
+      * - eservices without descriptors
+      * - eservices with only one descriptor that is in Draft state
+      */
+    val isSupervisorFilter =
+      Some(
+        Filters.nor(
```

Comment:
may I ask you to double check the conditions?
If I didn't miss something I think for the following sets the eservice is returned while it should not:
- Producer with Security role when the eservice has no descriptors
- Producer with Security role when the eservice has only one descriptor in draft
- Consumer with Security role when the eservice has no descriptors
- Consumer with Security role when the eservice has only one descriptor in draft